### PR TITLE
proxy/grpcproxy/cache/store.go: fixed TODO: use marshalTo to reduce

### PR DESCRIPTION
**Pull request by vimalk78**
modified proxy/grpcproxy/cache/store.go function keyFunc to use marshalTo instead of marshal

### CODE REVIEW

File: proxy/grpcproxy/cache/store.go [hunk 1]
```diff
@@ -25,8 +25,12 @@ import (
...
 
 var (
-	DefaultMaxEntries = 2048
-	ErrCompacted      = rpctypes.ErrGRPCCompacted
+	DefaultMaxEntries   = 2048
+	ErrCompacted        = rpctypes.ErrGRPCCompacted
+	MaxRangeRequestSize = 1024 * 4
+	// a temporary space used for Marshalling the RangeRequest for cache keyFunc
+	// should be used in keyFunc function only
+	sliceRangeRequestMarshal = make([]byte, MaxRangeRequestSize)
```
Comment (heyitsanthony) [inline proxy/grpcproxy/cache/store.go]: this should be in `cache`, otherwise there will be a data race if there are two concurrent `cache`s

**Comment by heyitsanthony** [comment]
did this make a performance difference? if not, should probably avoid premature optimization and keep it simple

**Comment by vimalk78** [comment]
i have not measured the performance, but since access to the sliceRangeRequestMarshal is synchronized, multiple routines cannot marshal the RangeRequests independently, there can be a dip in performance.

**Comment by vimalk78** [comment]
@xiang90  @heyitsanthony what should be done about this.

File: proxy/grpcproxy/cache/store.go [hunk 2]
```diff
@@ -62,15 +69,19 @@ type cache struct {
 	cachedRanges adt.IntervalTree
 
 	compactedRev int64
+	// a temporary space used for Marshalling the RangeRequest for cache keyFunc
+	// should be used in keyFunc function only
+	sliceRangeRequestMarshal []byte
```
Comment (xiang90) [inline proxy/grpcproxy/cache/store.go]: use sync.Pool?

**Pull request closed without merging**